flyteidl/protos/flyteidl/plugins/ray.proto

```diff
@@ -35,6 +37,8 @@ message HeadGroupSpec {
   // Optional. RayStartParams are the params of the start command: address, object-store-memory.
   // Refer to https://docs.ray.io/en/latest/ray-core/package-ref.html#ray-start
   map<string, string> ray_start_params = 1;
+  // Resource specification for ray head pod
+  core.Resources resources = 2;
```

Comment:
I'm considering whether we should use [CommonReplicaSpec](https://github.com/flyteorg/flyte/blob/13b3d82c94fb61047a49aa2093a2b17fb6a7c792/flyteidl/protos/flyteidl/plugins/common.proto#L15) here since someone might also want to set different image for the head node.

Reply:
I was considering similar ideas, having some sort of message to group these kinds of configurations but just figured the abstraction will come in time when use cases come along and the plugin evolves.  The `CommonReplicaSpec` could work but also includes fields like `replicas` that don't make sense in this context since there is only supposed to be a single head pod.

Reply:
I think if we wanted true flexibility we'd have some sort of `PodTemplateSpec` message since that would align with the kuberay CRDs. We'd just have to figure out how to merge those configurations with what Flyte is setting in the plugin. I'm willing to go down that route though..

Reply:
cc @EngHabu @hamersaw any thoughts on this

Reply:
We could start this off as 

```
message RayPodTemplateSpec {
  RayPodSpec pod_spec = 1;
}

message RayPodSpec {
  repeated RayContainer containers = 1;
}

message RayContainer {
  core.Resources resources = 1;
}
```

and add functionality as necessary.

Reply:
Ray supports a full PodTemplateSpec, we should not piece meal this. Just expose the full PodTemplate in the idl (maybe just reference [this](https://github.com/unionai/flyte/blob/f448a0358d8706a09b65b96543134f629327d755/flyteidl/protos/flyteidl/core/tasks.proto#L165) and let the SDK fill it in as needed. It's fine to expose a nicer interface in the SDK that takes the existing Resources object... etc. and populate the template... IMHO

Reply:
@EngHabu Looks like you linked to a private repo but if the line numbers are the same as the open source stuff I think it's the k8s pod spec which makes sense.. I'll update the PR